Mathlib/Data/Sum/Interval.lean

```diff
@@ -363,5 +404,53 @@ lemma Ioo_inr_inr :
     Ioo (inrₗ b₁ : α ⊕ₗ β) (inrₗ b₂) = (Ioo b₁ b₂).map (Embedding.inr.trans toLex.toEmbedding) := by
   rw [← Finset.map_map]; rfl
 
+end LocallyFiniteOrder
+
+section LocallyFiniteOrderBot
+variable [Preorder α] [Preorder β] [Fintype α] [LocallyFiniteOrderBot α] [LocallyFiniteOrderBot β]
+
+instance locallyFiniteOrderBot : LocallyFiniteOrderBot (α ⊕ₗ β) where
+  finsetIic := Sum.elim
+    (Iic · |>.map (.trans .inl toLex.toEmbedding))
+    (fun x => Finset.univ.disjSum (Iic x) |>.map toLex.toEmbedding) ∘ ofLex
+  finsetIio := Sum.elim
+    (Iio · |>.map (.trans .inl toLex.toEmbedding))
+    (fun x => Finset.univ.disjSum (Iio x) |>.map toLex.toEmbedding) ∘ ofLex
+  finset_mem_Iic := by simp
+  finset_mem_Iio := by simp
```

Comment:
This creates a diamond:
```

example [Fintype α] [Preorder α] [Preorder β] [OrderBot α] [OrderBot β] [OrderTop α]
    [LocallyFiniteOrder α] [LocallyFiniteOrder β] :
    LocallyFiniteOrder.toLocallyFiniteOrderBot = instLocallyFiniteOrderBot (α := α) (β := β) := by
  rfl
````
can you think of a way to make it not do so? Or is `LocallyFiniteOrder.toLocallyFiniteOrderBot` ultimately a bad instance?

Reply:
I think it ultimately is a bad instance. I knew it when I wrote it, but it was a great way to avoid writing too much boilerplate at a time where it wasn't yet clear whether `LocallyFiniteOrder` was even the right design

Reply:
So to be clear, in your opinion this is ok to merge as is, given the TODO comment about the diamond?